src/audio/chain_dma.c

```diff
@@ -224,6 +236,45 @@ static size_t chain_get_transferred_data_size(const uint32_t out_read_pos, const
 	return buff_size - in_read_pos + out_read_pos;
 }
 
+#if CONFIG_IPC_MAJOR_4
+static void xrun_notif_msg_init(struct ipc_msg *msg_xrun, uint32_t resource_id, uint32_t event_type)
+{
+	struct ipc4_resource_event_data_notification *notif_data = msg_xrun->tx_data;
+
+	notif_data->resource_id = resource_id;
+	notif_data->event_type = event_type;
+	notif_data->resource_type = GATEWAY;
+	notif_data->dws[0] = 0;
+	notif_data->dws[1] = 0;
+	notif_data->dws[2] = 0;
+	notif_data->dws[3] = 0;
```

Comment:
`dws[]` has 6 elements, you set 4 of them to 0 here. Is the intention of this to make it deterministic? Do you need to zero out all of them? Use `memset()`?